routers/web/repo/pull.go

```diff
@@ -447,7 +447,13 @@ func PrepareViewPullInfo(ctx *context.Context, issue *models.Issue) *git.Compare
 	}
 
 	if headBranchExist {
-		ctx.Data["UpdateAllowed"], err = pull_service.IsUserAllowedToUpdate(pull, ctx.User)
+		b, err := models.GetProtectedBranchBy(pull.HeadRepoID, pull.HeadBranch)
+		if err != nil {
+			ctx.ServerError("GetProtectedBranchBy", err)
+			return nil
+		}
+		ctx.Data["UpdateByRebaseNotAllowed"] = b != nil
+		ctx.Data["UpdateAllowed"], err = pull_service.IsUserAllowedToUpdate(pull, ctx.User, false)
```

Comment:
This is a shame as the pull_service.IsUserAllowedToUpdate will also get the ProtectedBranch.

I'm not sure what the best thing to do here is. It might be to make: 

`func IsUserAllowedToUpdate(pull *models.PullRequest, user *models.User) (mergeAllowed, rebaseAllowed bool, error)`

or else have a function that returns both.